querier.go

```diff
@@ -1210,3 +1398,85 @@ type errSeriesSet struct {
 func (s errSeriesSet) Next() bool { return false }
 func (s errSeriesSet) At() Series { return nil }
 func (s errSeriesSet) Err() error { return s.err }
+
+type chunkIterator struct {
+	chunks []chunks.Meta // series in time order
+	i      int
+}
+
+func (c *chunkIterator) Seek(t int64) bool {
+	if c.i >= len(c.chunks) {
+		return false
+	}
+
+	for c.Next() {
```

Comment:
Another call to Seek with the same timestamp is not noop here.

Also I don't see any tests for Seek for the chunk iterator.